qiita_ware/ebi.py

```diff
@@ -482,53 +403,81 @@ def generate_experiment_xml(self):
+                library_source = ET.SubElement(library_descriptor,
+                                               "LIBRARY_SOURCE")
+                library_source.text = sample_prep.pop('library_source')
+            if 'library_selection' in sample_prep:
+                library_selection = ET.SubElement(library_descriptor,
+                                                  "LIBRARY_SELECTION")
+                library_selection.text = sample_prep.pop('library_selection')
+            if 'library_layout' in sample_prep:
+                library_layout = ET.SubElement(library_descriptor,
+                                               "LIBRARY_LAYOUT")
+                library_layout.text = sample_prep.pop('library_layout')
+                # no idea what SINGLE means
+                ET.SubElement(library_layout, "SINGLE")
 
             self._generate_spot_descriptor(design, platform)
 
             platform_element = ET.SubElement(experiment, 'PLATFORM')
             platform_info = ET.SubElement(platform_element,
                                           platform.upper())
-            instrument_model = ET.SubElement(platform_info,
-                                             'INSTRUMENT_MODEL')
+            instrument_model = ET.SubElement(platform_info, 'INSTRUMENT_MODEL')
             instrument_model.text = 'unspecified'
 
-            if sample_info['prep']:
+            if sample_prep:
                 experiment_attributes = ET.SubElement(
                     experiment, 'EXPERIMENT_ATTRIBUTES')
                 self._add_dict_as_tags_and_values(experiment_attributes,
                                                   'EXPERIMENT_ATTRIBUTE',
-                                                  sample_info['prep'])
+                                                  sample_prep)
 
-        return experiment_set
+        return ET.tostring(experiment_set)
 
     def generate_run_xml(self):
         """Generates the run XML file
 
         Returns
         -------
-        xml.etree.Element
-            The root elelement of the generated ``ElementTree``
+        str
+            string with run XML values
         """
         run_set = ET.Element('RUN_SET', {
             "xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
             "xsi:noNamespaceSchemaLocation": "ftp://ftp.sra.ebi.ac.uk/meta/xsd"
                                              "/sra_1_3/SRA.run.xsd"})
-        for sample_name, sample_info in sorted(viewitems(self.samples)):
+        for sample_name, sample_prep in viewitems(self.samples_prep):
+            sample_prep = dict(sample_prep)
 
             experiment_alias = self._get_experiment_alias(sample_name)
 
-            file_type = sample_info['prep']['file_type']
-            file_path = sample_info['prep']['file_path']
+            # We only submit fastq
+            file_type = 'fastq'
+            file_path = join(self.ebi_dir, "%s.fastq.gz" % sample_name)
```

Comment:
I personally don't like this. The sample fastq file has been generated somewhere else, which means that the code to generate the filepath is duplicated. Should this be stored in some EBI attribute so we just query it instead of re-generating again here? The main concern is that if that code changes, we will need to change it in 2 places